script.rb

```diff
@@ -0,0 +1,176 @@
+# rubocop:disable all
+
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    var = to_a 
+    return to_enum(:my_each) unless block_given?
+    index = 0
+    while index < var.length
+      yield(var[index])
+      index += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    var = to_a
+    return to_enum(:my_each) unless block_given?
+
+    index = 0
+    while index < var.length
+      yield(var[index], index)
+      index += 1
+    end
+    self
+  end
+
+  def my_select
+    var = to_a
+    new_arr = []
+    return to_enum(:my_select) unless block_given?
+
+    var.my_each do |x|
+      new_arr.push(x) if yield(x) == true
+    end
+    new_arr
+  end
+
+  def my_all?(pattern = nil)
+    var = to_a
+    if block_given? == true
```

Comment:
- [ ] **OPTIONAL** You can leave it as `if block_given?` because that method already evaluates to true/false. Also applies to line 59 and 76